packages/jaeger-ui/src/components/DependencyGraph/JaegerForceGraphArrowLink.js

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import { ForceGraphLink} from 'react-vis-force';
+
+ function linkId(link) {
+  return `${link.source.id || link.source}=>${link.target.id || link.target}`;
+}
+
+
+export default class JaegerForceGraphArrowLink extends PureComponent { 
+  static get propTypes() {
+    return {
+      link: PropTypes.shape({
+  source: PropTypes.string.isRequired,
+  target: PropTypes.string.isRequired,
+  value: PropTypes.number,
+}).isRequired,
+      targetRadius: PropTypes.number,
+      edgeOffset: PropTypes.number,
+      className: PropTypes.string,
+      opacity: PropTypes.number,
+      stroke: PropTypes.string,
+      strokeWidth: PropTypes.number
+    };
+  }
```

Comment:
As we're using TypeScript, can you use a type instead of `prop-types`? E.g.

```tsx
type TProps = {
  link: {
    source: string;
    target: string;
    value: string;
  };
  targetRadius: number;
  // etc
};

export default class JaegerForceGraphArrowLink extends PureComponent<TProps> {
  // ...
```